src/annotator/anchoring/text-range.ts

```diff
@@ -32,33 +28,37 @@ function previousSiblingsTextLength(node) {
 }
 
 /**
- * Resolve one or more character offsets within an element to (text node, position)
- * pairs.
+ * Resolve one or more character offsets within an element to (text node,
+ * position) pairs.
  *
- * @param {Element} element
- * @param {number[]} offsets - Offsets, which must be sorted in ascending order
- * @return {{ node: Text, offset: number }[]}
+ * @param element
+ * @param offsets - Offsets, which must be sorted in ascending order
+ * @throws {RangeError}
  */
-function resolveOffsets(element, ...offsets) {
+function resolveOffsets(
+  element: Element,
+  ...offsets: number[]
+): { node: Text; offset: number }[] {
   let nextOffset = offsets.shift();
-  const nodeIter = /** @type {Document} */ (
-    element.ownerDocument
-  ).createNodeIterator(element, NodeFilter.SHOW_TEXT);
+  const nodeIter = element.ownerDocument.createNodeIterator(
+    element,
+    NodeFilter.SHOW_TEXT
+  );
   const results = [];
 
-  let currentNode = nodeIter.nextNode();
+  let currentNode = nodeIter.nextNode() as Text;
   let textNode;
   let length = 0;
 
   // Find the text node containing the `nextOffset`th character from the start
   // of `element`.
   while (nextOffset !== undefined && currentNode) {
-    textNode = /** @type {Text} */ (currentNode);
+    textNode = currentNode;
```

Comment:
I was curious about the need to assign `currentNode` to a separate reference here.

Reply:
On the last loop iteration, `currentNode` gets to `null` but the last non-null value of `currentNode` needs to be retained.